ModuleConcatenationPlugin throws on array default exports with functions

<!-- Please don't delete this template or we'll close your issue -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->
<!-- Also consider trying the webpack@beta version, maybe it's already fixed. -->

**Do you want to request a *feature* or report a *bug*?**
I want to report a bug.

<!-- Please ask questions on StackOverflow or the webpack Gitter (https://gitter.im/webpack/webpack). -->
<!-- Issues which contain questions or support requests will be closed. -->

**What is the current behavior?**
`webpack.optimize.ModuleConcatenationPlugin()` throws `Cannot read property 'range' of null` when two modules are imported that:
- declare a function with the same name
- declare a default export
- the default export is an array with empty (not undefined) elements

For instance, this will break:
```
function foo(n) {
  return 'bar';
}
export default [, foo];
```

But this is fine:
```
function foo(n) {
  return 'bar';
}
export default [undefined, foo];
```

**If the current behavior is a bug, please provide the steps to reproduce.**

<!-- A great way to do this is to provide your configuration via a GitHub gist. -->
<!-- Best provide a minimal reproduceable repo -->
<!-- If your issue is caused by a plugin or loader file the issue on the plugin/loader repo -->

I have a minimal repro at https://github.com/filipesilva/webpack-default-exports-concat that depends only on webpack.

```
git clone https://github.com/filipesilva/webpack-default-exports-concat
cd webpack-default-exports-concat
npm i
npm test
```

**What is the expected behavior?**
```
$ npm test

> webpack-default-exports@1.0.0 test D:\sandbox\webpack-default-exports
> webpack

Hash: 677662c2dee177954e76
Version: webpack 3.10.0
Time: 79ms
   [0] ./main.js + 2 modules 274 bytes {0} [built]

ERROR in chunk main [entry]
bundle.js
Cannot read property 'range' of null
npm ERR! Test failed.  See above for more details.
```

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
No browser, node v8.9.1, webpack 3.10.0, Windows 10

Related to https://github.com/angular/angular/issues/21809, https://github.com/webpack/webpack/issues/5415.

This bug also exists with `webpack@4.0.0-beta.0`:
```
$ npm test

> webpack-default-exports@1.0.0 test D:\sandbox\webpack-default-exports
> webpack

(node:3352) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Hash: 82ba7dd94993235c3dd2
Version: webpack 4.0.0-beta.0
Time: 105ms
Entrypoint main =
   [0] ./main.js + 2 modules 274 bytes {0} [built]
       | ./main.js 76 bytes [built]
       | ./import-one.js 99 bytes [built]
       | ./import-two.js 99 bytes [built]

WARNING in configuration
The 'mode' option has not been set. Set 'mode' option to 'development' or 'production' to enable defaults for this environment.

ERROR in chunk main [entry]
bundle.js
Cannot read property 'range' of null
npm ERR! Test failed.  See above for more details.
```

--- a/lib/optimize/ConcatenatedModule.js
+++ b/lib/optimize/ConcatenatedModule.js
@@ -143,6 +143,7 @@ function getPathInAst(ast, node) {
 	}
 
 	function enterNode(n) {
+		if(!n) return undefined;
 		const r = n.range;
 		if(r) {
 			if(r[0] <= nr[0] && r[1] >= nr[1]) {

--- a/test/cases/scope-hoisting/issue-6407/import-one.js
+++ b/test/cases/scope-hoisting/issue-6407/import-one.js
@@ -1,0 +1,4 @@
+function foo(n) {
+  return 'bar';
+}
+export default [, foo];

--- a/test/cases/scope-hoisting/issue-6407/import-two.js
+++ b/test/cases/scope-hoisting/issue-6407/import-two.js
@@ -1,0 +1,4 @@
+function foo(n) {
+  return 'bar';
+}
+export default [, foo];

--- a/test/cases/scope-hoisting/issue-6407/index.js
+++ b/test/cases/scope-hoisting/issue-6407/index.js
@@ -1,0 +1,11 @@
+import importOne from './import-one';
+import importTwo from './import-two';
+
+it("should concatenate modules default exports and empty array values", function() {
+	importOne.length.should.be.eql(2);
+	(typeof importOne[0]).should.be.eql('undefined');
+	(typeof importOne[1]).should.be.eql('function');
+	importTwo.length.should.be.eql(2);
+	(typeof importTwo[0]).should.be.eql('undefined');
+	(typeof importTwo[1]).should.be.eql('function');
+});

